syntax = "proto3"; // 声明版本

package mprpc; // 声明代码所在的包（对 C++ 来说是 namespace）

option cc_generic_services = true; // 表示生成 service 服务类和 rpc 方法描述，默认不开启

message LoginRequest {
    bytes name = 1;
    bytes pwd = 2;
}

message ResultCode {
    int32 errcode = 1;
    bytes errmsg = 2;
}

message LoginResponse {
    ResultCode result = 1;
    bool success = 2;
}

message GetFriendListRequest {
    uint32 userid = 1;
}

message User {
    bytes name = 1;
    uint32 age = 2;
    enum Sex {
        MAN = 0;
        WOMAN = 1;
    }
    Sex sex = 3;
}

message GetFriendListResponse {
    ResultCode result = 1;
    repeated User friend_list = 2; // 列表类型
}

service UserServiceRpc {
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetFriendList(GetFriendListRequest) returns(GetFriendListResponse);
}